File: LeetCode_Practice/Java_Solutions/Arrays/_121_BestTimeToBuySellStock.java
package LeetCode_Practice.Java_Solutions.Arrays;

public class _121_BestTimeToBuySellStock {
    public int maxProfit(int[] prices) {
        int lower = Integer.MAX_VALUE;
        int profit = 0, currProfit = 0;
        for (int i=0; i < prices.length; i++) {
            if (prices[i] < lower)
                lower = prices[i];
            currProfit = prices[i] - lower;
            if (profit < currProfit)
                profit = currProfit;
        }
        return profit;
    }
}

File: LeetCode_Practice/C++_Solutions/Arrays/_121_BestTimetoBuySellStock.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int low = INT_MAX, maxProfit = 0, currProfit = 0;
        for (int i=0; i < prices.size(); i++)
        {
            if (prices[i] < low)
                low = prices[i];

            currProfit = prices[i] - low;
            if (currProfit > maxProfit)
                maxProfit = currProfit;
        }
        return maxProfit;
    }
};